src/lib/sql/sql_translator.cpp

```diff
@@ -1799,16 +1803,33 @@ std::shared_ptr<AbstractExpression> SQLTranslator::_translate_hsql_expr(
 
     case hsql::kExprCast: {
       const auto source_data_type = left->data_type();
-      if (expr.columnType.data_type == hsql::DataType::DATE) {
-        AssertInput(source_data_type == DataType::String, "Cannot cast " + left->as_column_name() + " as Date");
-        // We do not know if an expression to be casted other than a ValueExpression actually contains dates, and we
-        // cannot check this later due to the lack of a Date data type
-        AssertInput(left->type == ExpressionType::Value, "Only ValueExpressions can be casted as Date");
-        const auto date_string = boost::get<pmr_string>(static_cast<ValueExpression&>(*left).value);
-        const auto date = string_to_date(std::string{date_string});
-        if (date) return std::const_pointer_cast<AbstractExpression>(left);
-        FailInput("'" + std::string{date_string} + "' is not a valid date");
+      const auto target_hsql_data_type = expr.columnType.data_type;
+
+      if (target_hsql_data_type == hsql::DataType::DATE || target_hsql_data_type == hsql::DataType::DATETIME) {
+        AssertInput(source_data_type == DataType::String,
+                    "Cannot cast " + left->as_column_name() + " as " +
+                        std::string{magic_enum::enum_name(target_hsql_data_type)});
```

Comment:
Isn't enum_name already returning an std::string?

Reply:
It's a [string view](https://github.com/Neargye/magic_enum/blob/master/doc/reference.md#enum_name) and iirc we have to create a string from it to use the `+` operator.